src/faillog.c

```diff
@@ -198,7 +198,7 @@ static bool reset_one (uid_t uid)
 	struct faillog fl;
 
 	offset = (off_t) uid * sizeof (fl);
```

Comment:
That cast could go away, if I'm not missing something.  Whatever `uid_t` is, it will be converted to `size_t`.  The same as with `off_t`.

Reply:
This is subject to Usual arithmetic conversions: <https://port70.net/%7Ensz/c/c11/n1570.html#6.3.1.8>

More precisely, the rule that will take effect here is:

> Otherwise, if the operand that has unsigned integer type has rank greater or equal to the rank of the type of the other operand, then the operand with signed integer type is converted to the type of the operand with unsigned integer type. 

`size_t` is always at least as wide as `uid_t` (usually wider), whatever the signedness of `uid_t` is.  So `uid` will be converted to `size_t` before the multiplication is performed.

Since `off_t` and `size_t` will most likely have the same width (because we're not using `_FILE_OFFSET_BITS`), `off_t` will also be converted to `size_t` before the multiplication is performed.

Reply:
> Since `off_t` and `size_t` will most likely have the same width (because we're not using `_FILE_OFFSET_BITS`), `off_t` would also be converted to `size_t` before the multiplication is performed.

See [configure.ac](https://github.com/shadow-maint/shadow/blob/03bbe6c4187d801057db6de24b47f309b12590e9/configure.ac#L54): Large file support is enabled by default for shadow.

Reply:
Ahh, my bad, that's why my grep didn't find anything.  It's autotools stuff :/  Then, yes, `off_t` will be the type to use, and we should cast.

Thanks!

Reply:
Added overflow checks